examples/upstream-tls.yaml

```diff
@@ -0,0 +1,51 @@
+kind: BackendPolicy
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: my-app
+spec:
+  backendRefs:
+  - name: my-service
+    group: core
+    kind: Service
+    port: 443
+  tls:
+    clientCertificateRef:
+      name: gateway-client-cert
+      group: core
+      kind: Secret
```

Comment:
It sounds like `clientCertificateRef` is moving to `Gateway`, so either file an issue to track the need to update this doc or update this PR based on the upcoming changes.